packages/light.js/src/frequency/utils/createPubsubObservable.ts

```diff
@@ -16,30 +17,30 @@ import { distinctReplayRefCount } from '../../utils/operators/distinctReplayRefC
  * @ignore
  * @example onAccountsChanged$, onEveryBlock$...
  */
-const createOnFromPubsub = <T>(
+const createPubsubObservable = <T>(
   pubsub: string,
-  api: any // TODO @parity/api
-): FrequencyObservable<T> => {
+  { provider }: FrequencyObservableOptions = {}
+) => {
   const [namespace, method] = pubsub.split('_');
+  const api = provider ? createApiFromProvider(provider) : getApi();
 
```

Comment:
we need to memoize at this point based on `pubsub` and `api`

Reply:
for example
```js
const createPubsubObservable = <T>(
  pubsub: string,
  { provider }: FrequencyObservableOptions = {}
) => {
  const [namespace, method] = pubsub.split('_');
  const api = provider ? createApiFromProvider(provider) : getApi();

  return memoize(createPubsubObservableWithApi)(pubsub, api);
};
```

and put the rest of the code in `createPubsubObservableWithApi`